lib/spack/spack/build_environment.py

```diff
@@ -302,6 +302,19 @@ def set_build_environment_variables(pkg, env, dirty=False):
             if '/macports/' in p:
                 env.remove_path('PATH', p)
 
+    # Set environment variables if specified for
+    # the given compiler
+    compiler = pkg.compiler
+    environment = compiler.environment
+    if 'set' in environment:
+        env_to_set = environment['set']
+        for key, value in env_to_set.iteritems():
+            env.set('SPACK_%s' % key, value)
```

Comment:
It's actually still important that these variables only be set *for the compilers*.  If the compiler depends on some external library, we don't want that to pollute the rest of the build.  If you put this in the `cc` wrapper, *only* the compiler sees the extra LD_LIBRARY_PATH.